torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp

```diff
@@ -222,6 +222,9 @@ bool guardDifferentiableGraph(Node* dnode) {
         }
       }
 
+      bool requires_grad = dni->type()->expectRef<TensorType>().requires_grad();
```

Comment:
nit: maybe add comment that because we are inserting a `RequiresGrad` check, the input's grad property is guaranteed, but other properties are not ?